assets/js/components/notifications/OfflineNotification.test.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * OfflineNotification component tests.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	render,
+	createTestRegistry,
+	waitFor,
+	act,
+} from '../../../../tests/js/test-utils';
+import { CORE_UI } from '../../googlesitekit/datastore/ui/constants';
+import OfflineNotification from './OfflineNotification';
+
+describe( 'InternalServerError', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	it( 'should display the notification when offline', async () => {
+		registry.dispatch( CORE_UI ).setValue( 'isOnline', false );
+
+		const { container, waitForRegistry } = render(
```

Comment:
I used `queryByText` to be able to use for checking if it is not in the document as well, to avoid error thrown by `getByText` when text is not present